arch/x86/x86_features.c

```diff
@@ -57,6 +59,16 @@ static void cpuidex(int info, int subinfo, unsigned* eax, unsigned* ebx, unsigne
 #endif
 }
 
+static uint64_t xgetbv(unsigned int xcr) {
+#ifdef _MSC_VER
+    return _xgetbv(xcr);
+#else
+    uint32_t eax, edx;
+    __asm__ ( ".byte 0x0f, 0x01, 0xd0" : "=a"(eax), "=d"(edx) : "c"(xcr));
+    return (uint64_t)(edx) << 32 | eax;
```

Comment:
GCC versions before 9.x series either don't have `_xgetbv` intrinsic or it is known to be buggy. I'm not sure if we want to use version check to decide whether to use the intrinsic if working and fallback to inline assembly when not available or broken.

Reply:
I haven't encountered anything previously that uses `_xgetbv` in GCC builds, so I think this might not be too much of an issue.

Reply:
And things would get really ugly and complicated if we would start testing support for `xgetbv` mnemonic in assembler too...